modules/topicsFpdModule.js

```diff
@@ -73,6 +114,145 @@ export function processFpd(config, {global}, {data = topicsData} = {}) {
+        const {domain, topics, bidder} = data.segment;
+        const iframeTopicsData = getTopicsData(domain, topics)[0];
+        iframeTopicsData && storeInLocalStorage(bidder, iframeTopicsData);
+      }
+    } catch (err) { }
+  }
+}
+
+/**
+Function to store Topics data recieved from iframe in storage(name: "prebid:topics")
+* @param {Topics} topics
+*/
+export function storeInLocalStorage(bidder, topics) {
+  const storedSegments = new Map(safeJSONParse(coreStorage.getDataFromLocalStorage(topicStorageName)));
+  if (storedSegments.has(bidder)) {
+    storedSegments.get(bidder)[topics['ext']['segclass']] = topics;
+    storedSegments.get(bidder)[lastUpdated] = new Date().getTime();
+    storedSegments.set(bidder, storedSegments.get(bidder));
+  } else {
+    storedSegments.set(bidder, {[topics.ext.segclass]: topics, [lastUpdated]: new Date().getTime()})
+  }
+  coreStorage.setDataInLocalStorage(topicStorageName, JSON.stringify([...storedSegments]));
+}
+
+function isCachedDataExpired(storedTime, cacheTime) {
+  const _MS_PER_DAY = 1000 * 60 * 60 * 24;
+  const now = new Date().getTime();
+  const daysDifference = Math.floor((storedTime - now) / _MS_PER_DAY);
+  return daysDifference > cacheTime;
+}
+
+/**
+* Function to get random bidders based on count passed with array of bidders
+**/
+function getRandomBidders(arr, count) {
+  return ([...arr].sort(() => 0.5 - Math.random())).slice(0, count)
+}
+
+/**
+ * function to add listener for message receiving from IFRAME
+ */
+function listenMessagesFromTopicIframe() {
+  window.addEventListener('message', receiveMessage, false);
+}
+
+function checkTCFv2(vendorData, requiredPurposes = TCF_REQUIRED_PURPOSES) {
+  const {gdprApplies, purpose} = vendorData;
+  if (!gdprApplies) {
+    return true;
+  }
+  return requiredPurposes.map((purposeNo) => {
+    const purposeConsent = purpose.consents ? purpose.consents[purposeNo] : false;
+    if (purposeConsent) {
+      return true;
+    }
+    return false;
+  }).reduce((a, b) => a && b, true);
+}
+
+function hasGDPRConsent() {
+  // Check for GDPR consent for purpose 1,2,3,4 and return false if consent has not been given
+  const gdprConsent = gdprDataHandler.getConsentData();
+  const hasGdpr = (gdprConsent && typeof gdprConsent.gdprApplies === 'boolean' && gdprConsent.gdprApplies) ? 1 : 0;
+  const gdprConsentString = hasGdpr ? gdprConsent.consentString : '';
+  if (hasGdpr) {
+    if ((!gdprConsentString || gdprConsentString === '') || !gdprConsent.vendorData) {
+      return false;
+    }
+    return checkTCFv2(gdprConsent.vendorData);
+  }
+  return true;
+}
+
+/**
+ * function to load the iframes of the bidder to load the topics data
+ */
+function loadTopicsForBidders() {
+  HAS_GDPR_CONSENT = hasGDPRConsent();
+  if (!HAS_GDPR_CONSENT) {
+    logInfo('Topics Module : Consent string is required to fetch the topics from third party domains.');
+    return;
+  }
```

Comment:
righty here can you use this function to check for dev access https://github.com/prebid/Prebid.js/blob/21712ca1f99e573c51d217ddcc4a4f08273dfb9d/src/utils.js#L820

Reply:
@patmmccann Device access check added.